### PR TITLE
Music Player improvements + JWPlayer updates integration

### PR DESCRIPTION
**Pull request by iisa**
This includes fixes specifically for these:
- Music Player, content window: background photo must hold aspect ratio
- Make pagination buttons round
- Music Player: on page load, click on highlighted track starts IA music player

JWPlayer Play8.js plumbing for:
- waveform toggle (only show waveform if no background photo is available)
- using Play8.js to set url so changing tracks between channels stay in sync

PLUS:
- data handling improvements

### CODE REVIEW

File: packages/ia-components/sandbox/theatres/components/audio-player/players_by_type/archive-audio-jwplayer-wrapper.jsx [hunk 1]
```diff
@@ -34,16 +36,32 @@ class ArchiveAudioPlayer extends Component {
...
 
     if (manuallyJumpToTrack) {
-      return player.playN(index);
+      stateUpdate.playerPlaylistIndex = index;
+      postStateCB = () => {
+        player.playN(index);
+      }
+    }
+
+    if (Object.keys(stateUpdate).length) {
```
Comment (bfalling) [inline packages/ia-components/sandbox/theatres/components/audio-player/players_by_type/archive-audio-jwplayer-wrapper.jsx]: Optional change but AirBNB JS advocates for explicit comparisons for strings and numbers.

File: packages/ia-components/sandbox/theatres/components/audio-player/players_by_type/archive-audio-jwplayer-wrapper.jsx [hunk 2]
```diff
@@ -105,11 +124,31 @@ class ArchiveAudioPlayer extends Component {
...
+   *
+   * Currently, this is where we support external ability to set URL
+   * through Internet Archive's JWPlayer Wrapper
+   */
+  postRegistration() {
+    const { onRegistrationComplete, jwplayerInfo, needsURLSettingAccess = false } = this.props;
+    const { identifier = '' } = jwplayerInfo;
+
+    if (onRegistrationComplete && needsURLSettingAccess) {
+      const externallySyncURL = function externallySyncURL (identifier, trackNumber) {
```
Comment (bfalling) [inline packages/ia-components/sandbox/theatres/components/audio-player/players_by_type/archive-audio-jwplayer-wrapper.jsx]: SUGGESTION: Remove space after function name.

File: packages/ia-components/sandbox/theatres/components/audio-player/players_by_type/archive-audio-jwplayer-wrapper.jsx [hunk 3]
```diff
@@ -105,11 +124,31 @@ class ArchiveAudioPlayer extends Component {
...
+        const playlistIndex = trackNumber - 1 || 0;
```
Comment (bfalling) [inline packages/ia-components/sandbox/theatres/components/audio-player/players_by_type/archive-audio-jwplayer-wrapper.jsx]: SUGGESTION: I don't believe this Boolean operator will do anything. If `trackNumber` is 1, then it's `0 || 0` which is still... `0`.

File: packages/ia-components/sandbox/theatres/components/track-list/track-list.jsx [hunk 4]
```diff
@@ -87,16 +88,20 @@ class TheatreTrackList extends Component {
     } = this.props;
 
     if (!tracks.length) return <p className="no-tracks">No tracks to display.</p>;
+
+    const [ firstTrack = {} ] = tracks;
```
Comment (bfalling) [inline packages/ia-components/sandbox/theatres/components/track-list/track-list.jsx]: SUGGESTION: AirBNB suggests no spaces inside square brackets.

File: packages/ia-components/sandbox/theatres/components/track-list/track-list.jsx [hunk 5]
```diff
@@ -87,16 +88,20 @@ class TheatreTrackList extends Component {
...
+    const trackNumberToHighlight = selectedTrack || firstTrack.trackNumber || null;
```
Comment (bfalling) [inline packages/ia-components/sandbox/theatres/components/track-list/track-list.jsx]: QUESTION: Does the final `|| null` serve a useful purpose?

File: packages/ia-components/sandbox/theatres/with-youtube-spotify/audio-with-youtube-spotify-main.jsx [hunk 6]
```diff
@@ -68,7 +68,7 @@ class AudioPlayerWithYoutubeSpotify extends Component {
       albumData,
       tracklistToShow: [],
       channelToPlay: 'archive',
-      trackSelected: 1, /* 0 = album */
+      trackSelected: null, /* 0 = album */
```
Comment (bfalling) [inline packages/ia-components/sandbox/theatres/with-youtube-spotify/audio-with-youtube-spotify-main.jsx]: SUGGESTION: You note what value `0` means. What about its default value of `null`?

Comment (iisa) [reply packages/ia-components/sandbox/theatres/with-youtube-spotify/audio-with-youtube-spotify-main.jsx]: `null` signifies that nothing has been chosen yet by user.  This update is to integrate in fix for playing an item on first click.

File: packages/ia-components/sandbox/theatres/components/audio-player/players_by_type/archive-audio-jwplayer-wrapper.jsx (hunk 3 above)
Comment (iisa) [reply packages/ia-components/sandbox/theatres/components/audio-player/players_by_type/archive-audio-jwplayer-wrapper.jsx]: yes, agree. it was a defensive reflex.

File: packages/ia-components/sandbox/theatres/components/track-list/track-list.jsx (hunk 5 above)
Comment (iisa) [reply packages/ia-components/sandbox/theatres/components/track-list/track-list.jsx]: another defensive reflex.

**Pull request merged**